src/OrchardCore.Modules/OrchardCore.Tenants/Controllers/AdminController.cs

```diff
@@ -238,6 +247,11 @@ public async Task<IActionResult> Index(BulkActionViewModel model)
                 return Forbid();
             }
 
+            if (model.BulkAction.ToString() == "Remove" && !_tenantsOptions.TenantRemovalAllowed)
+            {
+                return Forbid();
```

Comment:
I'd display a notification instead of forbid here, to be clear what's happening.

Reply:
In that case the "Remove" button is no more rendered, so here only for security.

Let me know if you still want the notification.

Reply:
The button is still there when 'TenantRemovalAllowed' is false.

Reply:
Yes, I only removed it today, 2 commits ago

Reply:
I see. Then this can remain, no need for a notification, then.